src/convert.ts

```diff
@@ -1007,112 +1008,92 @@ export default function convert(config: ConvertConfig): ESTreeNode | null {
       } else if (
         !(result as any).static &&
         node.name.kind === SyntaxKind.StringLiteral &&
-        node.name.text === 'constructor'
+        node.name.text === 'constructor' &&
+        result.type !== AST_NODE_TYPES.Property
       ) {
         (result as any).kind = 'constructor';
       }
 
       // Process typeParameters
       if (node.typeParameters && node.typeParameters.length) {
-        (method as any).typeParameters = convertTSTypeParametersToTypeParametersDeclaration(
-          node.typeParameters
-        );
+        if (result.type !== AST_NODE_TYPES.Property) {
+          method.typeParameters = convertTSTypeParametersToTypeParametersDeclaration(
+            node.typeParameters
+          );
+        } else {
+          result.typeParameters = convertTSTypeParametersToTypeParametersDeclaration(
+            node.typeParameters
+          );
+        }
```

Comment:
babel parses it in this way, in objects `typeParameters` are assigned to Property, and in classes to methods